src/platform/i386/entry.S

```diff
@@ -72,29 +72,13 @@ PT_OLDSS	= 0x40
 /* we are changing thread contexts, reload all registers */
 .align 16
 restore_from_thd:
-	pushl %ecx
-	pushl %edx
-	call cos_get_curr_thd_asm
-	popl %edx
-	popl %ecx
-
-	addl $THD_REGS, %eax /* offsetof(struct thread, regs) */
-	/* restore from preemption */
-	movl PT_EBX(%eax), %ebx
-	movl PT_ECX(%eax), %ecx
-	movl PT_EDX(%eax), %edx
-	movl PT_ESI(%eax), %esi
-	movl PT_EDI(%eax), %edi
-	movl PT_EBP(%eax), %ebp
-
-	/* set up the activation record required by iret */
-	pushl $(SEL_UDSEG)
-	pushl PT_OLDESP(%eax)
-	pushl PT_EFLAGS(%eax)
-	pushl $(SEL_UCSEG)
-	pushl PT_EIP(%eax)
-
-	movl PT_EAX(%eax), %eax
+	RESTORE_REGS_GENERAL
```

Comment:
Nice, you got this working!!!! awesome.  Looks much better.  I believe we can remove `cos_get_curr_thd_asm` now.  Check that it has no additional uses, and remove it if you can.